gax/src/main/java/com/google/api/gax/tracing/NoopApiTracerFactory.java

```diff
@@ -36,15 +36,15 @@
  *
  * <p>For internal use only.
  */
-@InternalApi
-public final class NoopApiTracerFactory implements ApiTracerFactory {
+@InternalApi("For internal use by google-cloud-java clients only")
+public class NoopApiTracerFactory implements ApiTracerFactory {
   private static final NoopApiTracerFactory INSTANCE = new NoopApiTracerFactory();
 
   public static NoopApiTracerFactory getInstance() {
     return INSTANCE;
   }
 
-  private NoopApiTracerFactory() {}
+  public NoopApiTracerFactory() {}
```

Comment:
protected

Reply:
Hmm, if it's package private I can't extend it in the client library?

Reply:
Why would anyone need to extend this? 

If it's to create a non-Noop tracer factory then it violates the principle that the subclass IS A NoopApiTracerFactory.

Reply:
We want to add more methods to `ApiTracer` (and `ApiTracerFactory`), it'll be a breaking change if client is implementing it. 

So we're marking `ApiTracer` to be internal, and update client code to extend `NoopApiTracer`, so whenever we add a new methods to `ApiTracer`, it'll have default implementation in `NoopApiTracer` and won't break any client code.

Reply:
not package private, but protected.

This is to workaround not having default methods in interfaces in java7. We want to be able to add new events that can be traced without breaking implementations of ApiTracer. In java8, we would just add noop methods in the interface. However in java7 we need a base class to stick the noop impl into

Reply:
That's reasonable but in that case you should not name this NoopTracerFactory because the subclasses aren't noops and therefore aren't NoopTracerFactories. TracerFactory or BaseApiTracerFactory or something like that would be OK.

Reply:
Renamed to `BaseApiTracer` and `BaseApiTracerFactory`